### PR TITLE
Deadlock in QuickInfo, again. Fixes #5254

### LINKED ISSUE #5254: Requesting Quick Info hangs VS
Please provide a succinct description of the issue.

#### Repro steps

In an .fsx script in VS 2017.7.4, when I hit CTRL K + I when the caret is over a function, VS hangs completely - no redraws occur, nothing. The only option is to restart it.

### PR DESCRIPTION
**Pull request by majocha**
This is the same issue and the same solution as #4713.
I've added some comments this time.

CC @cartermp

### CODE REVIEW

**Comment by cartermp** [comment]
@KevinRansom We'll need to take this as an urgent regression fix.

File: vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs [hunk 1]
```diff
@@ -160,12 +160,15 @@ module private FSharpQuickInfo =
 type internal FSharpAsyncQuickInfoSource
     (
         serviceProvider: IServiceProvider,
-        xmlMemberIndexService: IVsXMLMemberIndexService,
         checkerProvider:FSharpCheckerProvider,
         projectInfoManager:FSharpProjectOptionsManager,
         textBuffer:ITextBuffer
     ) =
 
+    // GetService calls must be made on the UI thread
```
Comment (vasily-kirichenko) [inline vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs]: It's unclear _why_ the following lines are always called from the UI thread though.

**Comment by vasily-kirichenko** [comment]
I think we have a couple more places in the codebase where we are not given services as ctor parameters, but obtain them from ServiceProvider, we should check if we do it in the UI thread as well.

**Comment by cartermp** [comment]
@vasily-kirichenko It's worth taking a look at, though it may be only relevant to QuickInfo today since it's now using an editor API that's called from a background thread. As the editor has more of the Roslyn stuff moved down into it, things like this will definitely have to be re-evaluated.

File: vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs (hunk 1 above)
Comment (majocha) [reply vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs]: Yes, I'll move it to TryCreateQuickInfoSource which should be safe according to https://github.com/Microsoft/visualfsharp/pull/4713#issuecomment-380566134

**Pull request merged**